### ISSUE TITLE
Indentation Autofix doesn't always apply

### ISSUE THREAD

**Issue by mwallace72** [nvuillam/npm-groovy-lint#121]
Example config
```
{
  "extends": "recommended-jenkinsfile",
  "rules": {
    "formatting.Indentation": {
      "spacesPerIndentLevel": 2,
      "severity": "error"
    }
  }
}
```

Example code
```
pipeline {
  agent any
  stages {
        stage('Intall JQ') {
      steps {
        script {
          sh 'apt update; apt install jq=1.5+dfsg-2+b1 -y'
        }
      }
        }
  }
}
```

Running with `--fix` does nothing to the file, nor are errors flagged with incorrect spacing in file.
If a line has _less_ than the required spaces for the indent level, it will be fixed.
```
pipeline {
agent any // <- this will be fixed
  stages {
        stage('Intall JQ') {
      steps {
        script {
          sh 'apt update; apt install jq=1.5+dfsg-2+b1 -y'
        }
      }
        }
  }
}
```

Additionally, if a file has a bad multiple of the required spaces for the indent level, it will be fixed.
```
pipeline {
  agent any
   stages { // <- this has 3 spaces
        stage('Intall JQ') {
      steps {
        script {
          sh 'apt update; apt install jq=1.5+dfsg-2+b1 -y'
        }
      }
        }
  }
}
```

**Comment by nvuillam** [comment]
The problem is that I rely on CodeNarc to know which fixes to perform, and its Indentation rule does not detect / provide what I need to perfectly format in this case :/
With CodeNarc 2.0, it gets (a little) better.

Before --fix:
```groovy
pipeline {
agent any
        stages {
            stage('Intall JQ') {
            steps {
            script {
                sh 'apt update; apt install jq=1.5+dfsg-2+b1 -y'
            }
            }
            }
        }
    }
```

after fix:

```groovy
pipeline {
    agent any
        stages {
            stage('Intall JQ') {
            steps {
                script {
                    sh 'apt update; apt install jq=1.5+dfsg-2+b1 -y'
                }
            }
            }
        }
}
```

So... far from perfect, but I can't do more for now 😢 

You can test it with beta version : `npm i npm-groovy-lint@8.0.0-beta.4`
